packages/appsync-emulator-serverless/__test__/schemaTest.test.js

```diff
@@ -314,4 +314,137 @@ describe('creates executable schema', () => {
 
     expect(result).toMatchObject({ data: { jsonTest: { test: 'yup' } } });
   });
+
+  it('AWSDate scalar', async () => {
+    const field = 'dateTest';
+    const val = new Date('05 October 2011').toISOString().split('T')[0];
+
+    const source = `
+      query {
+        ${field}(${field}: "${val}")
+      }
+    `;
```

Comment:
To be honest, I'm not crazy about doing:

```
const source = `
      query {
        ${field}(${field}: "${val}")
      }
    `;
```

and

```
expect(result).toMatchObject({ data: { [field]: val } });
```

In every test.  I would prefer each test query to be explicit and readable:

```
query {
  dateTest(dateTest: new Date('05 October 2011').toISOString().split('T')[0])
}
```

and

```
expect(result).toMatchObject({ data: { dateTest: new Date('05 October 2011').toISOString().split('T')[0]  } });
```

However, I appreciate you taking the time to work on this project.  Perhaps as a middle ground we can do something like:

```
getScalarSource  = (field, val) => `
  query {
      ${field}(${field}: "${val}")
  }
`
```
and

```
expectScalarResult = (result,field,val) => expect(result).toMatchObject({ data: { [field]: val } })
```

and define these methods in an outer scope?

Let me know, ready to land this into master. 👍

Reply:
The latest commit implements this approach.